pycbc/waveform/ringdown.py

```diff
@@ -323,60 +373,119 @@ def td_output_vector(freqs, damping_times, taper=False,
+    azimuthal : float, optional
+        The azimuthal angle. Only used if harmonics is 'spherical' or
+        'spheroidal'. Default is 0.
+    spin : float, optional
+        The dimensionless spin of the black hole. Must be provided if
+        harmonics is 'spheroidal'. Ignored otherwise.
+    pol : float, optional
+        Angle (in radians) to use for arbitrary "+m" harmonic. Must be provided
+        if harmonics is 'arbitrary'. Ignored otherwise.
+    polnm : float, optional
+        Angle (in radians) to use for arbitrary "-m" harmonic. Must be provided
+        if harmonics is 'arbitrary'. Ignored otherwise.
+
+    Returns
+    -------
+    xlm : complex
+        The harmonic of the +m mode.
+    xlnm : complex
+        The harmonic of the -m mode.
+    """
+    if harmonics == 'spherical':
+        xlm = lal.SpinWeightedSphericalHarmonic(inclination, azimuthal, -2,
+                                                l, m)
+        xlnm = lal.SpinWeightedSphericalHarmonic(inclination, azimuthal, -2,
+                                                 l, -m)
+    elif harmonics == 'spheroidal':
+        if spin is None:
+            raise ValueError("must provide a spin for spheroidal harmonics")
+        if pykerr is None:
+            raise ImportError("pykerr must be installed for spheroidal "
+                              "harmonics")
+        xlm = pykerr.spheroidal(inclination, spin, l, m, n, phi=azimuthal)
+        xlnm = pykerr.spheroidal(inclination, spin, l, -m, n, phi=azimuthal)
+    elif harmonics == 'arbitrary':
+        if pol is None or polnm is None:
+            raise ValueError('must provide a pol and a polnm for arbitrary '
+                             'harmonics')
+        xlm = numpy.exp(1j*pol)
+        xlnm = numpy.exp(1j*polnm)
+    else:
+        raise ValueError("harmonics must be either spherical, spheroidal, "
+                         "or arbitrary")
+    return xlm, xlnm
 
-    return Y_plus, Y_cross
 
 def Kerr_factor(final_mass, distance):
     """Return the factor final_mass/distance (in dimensionless units) for Kerr
     ringdowns
     """
-
     # Convert solar masses to meters
     mass = final_mass * lal.MSUN_SI * lal.G_SI / lal.C_SI**2
     # Convert Mpc to meters
     dist = distance * 1e6 * lal.PC_SI
-
     return mass / dist
 
+
 # Functions for tapering ######################################################
 
-def apply_taper(tau, amp, phi, delta_t, l=2, m=2, inclination=None):
+def apply_taper(tau, amp, phi, delta_t, l=2, m=2, inclination=None, pol=None):
     """Return a tapering window of the form exp(10*t/tau).
     """
-
+    raise ValueError("FIX ME")
     taper_times = -numpy.arange(0, int(tau/delta_t))[::-1] * delta_t
-    if inclination is not None:
+    if pol is not None:
+        Y_plus = numpy.cos(pol)
+        Y_cross = numpy.sin(pol)
+    elif inclination is not None:
         Y_plus, Y_cross = spher_harms(l, m, inclination)
     else:
         Y_plus, Y_cross = 1, 1
```

Comment:
To do: remove taper.

Reply:
I left the taper capabilities in, but fixed them. Tapering had been broken for quite some time.